Missing page_client-reference-manifest.js files on build (using path grouping)

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #22 SMP Tue Jan 10 18:39:00 UTC 2023
    Binaries:
      Node: 16.17.0
      npm: 8.15.0
      Yarn: 1.22.19
      pnpm: 7.1.0
    Relevant Packages:
      next: 13.4.13-canary.13
      eslint-config-next: 13.2.4
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 4.9.5
    Next.js Config:
      output: N/A
```


### Which area(s) of Next.js are affected? (leave empty if unsure)

App Router

### Link to the code that reproduces this issue or a replay of the bug

https://codesandbox.io/p/sandbox/nextjs-missing-manifest-rtgf6d

### To Reproduce

The error occurs when you try to build the application for the first time (when no .next folder exists).
Just running `npm run build` will produce the following output:

```
➜  workspace git:(master) ✗ npm run build

> next-13@0.1.0 build
> next build

Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
- info Creating an optimized production build  
- info Compiled successfully
- info Linting and checking validity of types .Pages directory cannot be found at /project/home/nickman87/workspace/pages or /project/home/nickman87/workspace/src/pages. If using a custom path, please configure with the `no-html-link-for-pages` rule in your eslint config file.
- info Linting and checking validity of types  
- info Collecting page data ...Error: Cannot find module '/project/home/nickman87/workspace/.next/server/app/(logged-in)/page_client-reference-manifest.js'
Require stack:
- /project/home/nickman87/workspace/node_modules/next/dist/server/load-components.js
- /project/home/nickman87/workspace/node_modules/next/dist/build/utils.js
- /project/home/nickman87/workspace/node_modules/next/dist/build/worker.js
- /project/home/nickman87/workspace/node_modules/next/dist/compiled/jest-worker/processChild.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
    at /project/home/nickman87/workspace/node_modules/next/dist/server/require-hook.js:110:36
    at Function.Module._load (node:internal/modules/cjs/loader:804:27)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at loadJSManifest (/project/home/nickman87/workspace/node_modules/next/dist/server/load-components.js:64:72)
    at loadComponentsImpl (/project/home/nickman87/workspace/node_modules/next/dist/server/load-components.js:86:29)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /project/home/nickman87/workspace/node_modules/next/dist/build/utils.js:1006:32
    at async Span.traceAsyncFn (/project/home/nickman87/workspace/node_modules/next/dist/trace/trace.js:103:20) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/project/home/nickman87/workspace/node_modules/next/dist/server/load-components.js',
    '/project/home/nickman87/workspace/node_modules/next/dist/build/utils.js',
    '/project/home/nickman87/workspace/node_modules/next/dist/build/worker.js',
    '/project/home/nickman87/workspace/node_modules/next/dist/compiled/jest-worker/processChild.js'
  ]
}

> Build error occurred
Error: Failed to collect page data for /
    at /project/home/nickman87/workspace/node_modules/next/dist/build/utils.js:1156:15
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  type: 'Error'
}
- info Collecting page data .                                                                             
```

### Describe the Bug

The build will generate the necessary file `page_client-reference-manifest.js` for the `(logged-out)` group, but not for the `(logged-in)` group folder.
This causes a one-time error during build, in subsequent builds the error does not seem to appear anymore, as well as a missing module error during runtime (as the file is loaded, but does not exist).

The issue appears starting 13.4.10, but it is not shown at compile time, only at runtime.

### Expected Behavior

All necessary files should be generated on build time.

### Which browser are you using? (if relevant)

Chrome, but not relevant

### How are you deploying your application? (if relevant)

Netlify, but the issue also arises locally

--- a/packages/next/src/build/webpack/loaders/next-app-loader.ts
+++ b/packages/next/src/build/webpack/loaders/next-app-loader.ts
@@ -462,6 +462,14 @@ const nextAppLoader: AppLoader = async function nextAppLoader() {
           continue
         }
 
+        // avoid clobbering existing page segments
+        // if it's a valid parallel segment, the `children` property will be set appropriately
+        if (matched.children && matched.children !== rest[0]) {
+          throw new Error(
+            `/You cannot have two parallel pages that resolve to the same path. Please check ${appPath}.`
+          )
+        }
+
         matched.children = rest[0]
       }
     }

--- a/test/e2e/app-dir/parallel-routes-and-interception/app/parallel/(new)/@baz/default.js
+++ b/test/e2e/app-dir/parallel-routes-and-interception/app/parallel/(new)/@baz/default.js
@@ -1,0 +1,3 @@
+export default function Page() {
+  return <div>parallel/@bar/nested/default</div>
+}

--- a/test/e2e/app-dir/parallel-routes-and-interception/app/parallel/@bar/default.js
+++ b/test/e2e/app-dir/parallel-routes-and-interception/app/parallel/@bar/default.js
@@ -1,0 +1,3 @@
+export default function Page() {
+  return <div>parallel/@bar/default</div>
+}

--- a/test/e2e/app-dir/parallel-routes-and-interception/app/parallel/@bar/nested/page.js
+++ b/test/e2e/app-dir/parallel-routes-and-interception/app/parallel/@bar/nested/page.js
@@ -1,0 +1,3 @@
+export default function Page() {
+  return <div>parallel/@bar/nested/page</div>
+}

--- a/test/e2e/app-dir/parallel-routes-and-interception/app/parallel/@bar/nested/page.js
+++ b/test/e2e/app-dir/parallel-routes-and-interception/app/parallel/@bar/nested/page.js
@@ -1,3 +1,3 @@
 export default function Page() {
-  return <div>parallel/@bar/nested/page</div>
+  return <div>parallel/@foo/nested/page</div>
 }

--- a/test/e2e/app-dir/parallel-routes-and-interception/app/parallel/@foo/default.js
+++ b/test/e2e/app-dir/parallel-routes-and-interception/app/parallel/@foo/default.js
@@ -1,0 +1,3 @@
+export default function Page() {
+  return <div>parallel/@foo/default</div>
+}

--- a/test/e2e/app-dir/parallel-routes-and-interception/app/parallel/@foo/nested/page.js
+++ b/test/e2e/app-dir/parallel-routes-and-interception/app/parallel/@foo/nested/page.js
@@ -1,0 +1,3 @@
+export default function Page() {
+  return <div>parallel/@foo/nested/page</div>
+}

--- a/test/e2e/app-dir/parallel-routes-and-interception/app/parallel/layout.js
+++ b/test/e2e/app-dir/parallel-routes-and-interception/app/parallel/layout.js
@@ -2,7 +2,7 @@ import './style.css'
 
 export default function Parallel({ foo, bar, children }) {
   return (
-    <div>
+    <div id="parallel-layout">
       parallel/layout:
       <div className="parallel" title="@foo">
         {foo}

--- a/test/e2e/app-dir/parallel-routes-and-interception/parallel-routes-and-interception.test.ts
+++ b/test/e2e/app-dir/parallel-routes-and-interception/parallel-routes-and-interception.test.ts
@@ -1,5 +1,6 @@
 import { createNextDescribe } from 'e2e-utils'
 import { check } from 'next-test-utils'
+import { outdent } from 'outdent'
 
 createNextDescribe(
   'parallel-routes-and-interception',
@@ -165,27 +166,57 @@ createNextDescribe(
         await step5()
       })
 
-      // FIXME: this parallel route test is broken, shouldn't only check if html containing those strings
-      // previous they're erroring so the html is empty but those strings are still in flight response.
-      it.skip('should match parallel routes', async () => {
-        const html = await next.render('/parallel/nested')
-        expect(html).toContain('parallel/layout')
-        expect(html).toContain('parallel/@foo/nested/layout')
-        expect(html).toContain('parallel/@foo/nested/@a/page')
-        expect(html).toContain('parallel/@foo/nested/@b/page')
-        expect(html).toContain('parallel/@bar/nested/layout')
-        expect(html).toContain('parallel/@bar/nested/@a/page')
-        expect(html).toContain('parallel/@bar/nested/@b/page')
-        expect(html).toContain('parallel/nested/page')
+      it('should match parallel routes', async () => {
+        const $ = await next.render$('/parallel/nested')
+        const pageText = $('#parallel-layout').text()
+        expect(pageText).toContain('parallel/layout')
+        expect(pageText).toContain('parallel/@foo/nested/layout')
+        expect(pageText).toContain('parallel/@foo/nested/@a/page')
+        expect(pageText).toContain('parallel/@foo/nested/@b/page')
+        expect(pageText).toContain('parallel/@bar/nested/layout')
+        expect(pageText).toContain('parallel/@bar/nested/@a/page')
+        expect(pageText).toContain('parallel/@bar/nested/@b/page')
+        expect(pageText).toContain('parallel/nested/page')
       })
 
-      // FIXME: this parallel route test is broken, shouldn't only check if html containing those strings
-      // previous they're erroring so the html is empty but those strings are still in flight response.
-      it.skip('should match parallel routes in route groups', async () => {
-        const html = await next.render('/parallel/nested-2')
-        expect(html).toContain('parallel/layout')
-        expect(html).toContain('parallel/(new)/layout')
-        expect(html).toContain('parallel/(new)/@baz/nested/page')
+      it('should match parallel routes in route groups', async () => {
+        const $ = await next.render$('/parallel/nested-2')
+        const pageText = $('#parallel-layout').text()
+        expect(pageText).toContain('parallel/layout')
+        expect(pageText).toContain('parallel/(new)/layout')
+        expect(pageText).toContain('parallel/(new)/@baz/nested/page')
+      })
+
+      it('should throw an error when a route groups causes a conflict with a parallel segment', async () => {
+        await next.stop()
+        await next.patchFile(
+          'app/parallel/nested-2/page.js',
+          outdent`
+              export default function Page() {
+                return 'hello world'
+              }
+            `
+        )
+
+        if (isNextDev) {
+          await next.start()
+
+          const html = await next.render('/parallel/nested-2')
+
+          expect(html).toContain(
+            'You cannot have two parallel pages that resolve to the same path.'
+          )
+        } else {
+          await expect(next.start()).rejects.toThrow('next build failed')
+
+          await check(
+            () => next.cliOutput,
+            /You cannot have two parallel pages that resolve to the same path./i
+          )
+        }
+        await next.stop()
+        await next.deleteFile('app/parallel/nested-2/page.js')
+        await next.start()
       })
 
       it('should throw a 404 when no matching parallel route is found', async () => {

